lib/content-services/src/lib/tag/tags-creator/tags-creator.component.ts

```diff
@@ -0,0 +1,379 @@
+
+        this.setTagNameControlErrorMessageKey();
+    }
+
+    ngOnDestroy(): void {
+        this.onDestroy$.next();
+        this.onDestroy$.complete();
+        this.cancelExistingTagsLoading$.next();
+        this.cancelExistingTagsLoading$.complete();
+    }
+
+    @HostBinding('class.adf-creator-with-existing-tags-panel')
+    get hostClass(): boolean {
+        return this.existingTagsPanelVisible;
+    }
+
+    get tagNameControl(): FormControl<string> {
+        return this._tagNameControl;
+    }
+
+    get existingTags(): TagEntry[] {
+        return this._existingTags;
+    }
+
+    get tagNameErrorMessageKey(): string {
+        return this._tagNameErrorMessageKey;
+    }
+
+    get existingTagsLoading(): boolean {
+        return this._existingTagsLoading;
+    }
+
+    get typing(): boolean {
+        return this._typing;
+    }
+
+    get tagsListScrollbarVisible(): boolean {
+        return this._tagsListScrollbarVisible;
+    }
+
+    get existingTagsPanelVisible(): boolean {
+        return this._existingTagsPanelVisible;
+    }
+
+    get existingTagsLabelKey(): string {
+        return this._existingTagsLabelKey;
+    }
+
+    hideNameInput(): void {
+        this.tagNameControlVisible = false;
+        this._existingTagsPanelVisible = false;
+        this.existingTagsPanelVisibilityChange.emit(this.existingTagsPanelVisible);
+        this.tagNameControlVisibleChange.emit(this.tagNameControlVisible);
+    }
+
+    addTag(): void {
+        if (!this._typing && !this.tagNameControl.invalid) {
+            this.tags.push(this.tagNameControl.value.trim());
+            this.hideNameInput();
+            this.tagNameControl.setValue('');
+            this.checkScrollbarVisibility();
+            this.tagNameControl.markAsUntouched();
+            this.tagsChange.emit(this.tags);
+        }
+    }
+
+    removeTag(tag: string): void {
+        this.removeTagFromArray(this.tags, tag);
+        this.tagNameControl.updateValueAndValidity();
+        this.updateExistingTagsListOnRemoveFromTagsToConfirm(tag);
+        this.checkScrollbarVisibility();
+        this.tagsChange.emit(this.tags);
+    }
+
+    addExistingTagToTagsToAssign(change: MatSelectionListChange): void {
+        const selectedTag: TagEntry = change.options[0].value;
+        this.tags.push(selectedTag.entry.tag);
+        this.removeTagFromArray(this.existingTags, selectedTag);
+        this.tagNameControl.updateValueAndValidity();
+        this.exactTagSet$.next();
+        this.tagsChange.emit(this.tags);
+    }
+
+    isOnlyCreateMode(): boolean {
+        return this.mode === TagsCreatorMode.CREATE;
+    }
+
+    private onTagNameControlValueChange(name: string): void {
+        this.tagNameControl.markAsTouched();
+        this.loadTags(name);
+    }
+
+    private loadTags(name: string) {
+        if (name) {
+            forkJoin({
+                exactResult: this.tagService.findTagByName(name),
```

Comment:
Do we need the exact result? According to what Tom said backend should already return most important results while searching and you could simplify `loadTags` then

Reply:
Yes, we need that. I checked this some time ago with Tom when implementing that in ACC and he said that he was mistaken regarding to this because he thought that backend works like returning exact value as first result but that's not true. Here you can see that case on screenshot, please have a look: 
<img width="1020" alt="image" src="https://user-images.githubusercontent.com/115619721/228157910-30003dcc-b0f8-4baa-b77b-bd8b6c3dbc24.png">

I typed test2 but in backend's response test2 is on second position, not on first unfortunately. So there may be a case that user completely won't see exact value if we don't load it separately.

Reply:
Ok no problem then